genesyscloud/outbound_contactlistfilter/genesyscloud_outbound_contactlistfilter_proxy.go

```diff
@@ -0,0 +1,193 @@
+func (p *outboundContactlistfilterProxy) updateOutboundContactlistfilter(ctx context.Context, id string, outboundContactlistfilter *platformclientv2.Contactlistfilter) (*platformclientv2.Contactlistfilter, error) {
+	return p.updateOutboundContactlistfilterAttr(ctx, p, id, outboundContactlistfilter)
+}
+
+// deleteOutboundContactlistfilter deletes a Genesys Cloud outbound contactlistfilter by Id
+func (p *outboundContactlistfilterProxy) deleteOutboundContactlistfilter(ctx context.Context, id string) (response *platformclientv2.APIResponse, err error) {
+	return p.deleteOutboundContactlistfilterAttr(ctx, p, id)
+}
+
+// createOutboundContactlistfilterFn is an implementation function for creating a Genesys Cloud outbound contactlistfilter
+func createOutboundContactlistfilterFn(ctx context.Context, p *outboundContactlistfilterProxy, outboundContactlistfilter *platformclientv2.Contactlistfilter) (*platformclientv2.Contactlistfilter, error) {
+	contactListFilter, _, err := p.outboundApi.PostOutboundContactlistfilters(*outboundContactlistfilter)
+	if err != nil {
+		return nil, err
+	}
+
+	return contactListFilter, nil
+}
+
+// getAllOutboundContactlistfilterFn is the implementation for retrieving all outbound contactlistfilter in Genesys Cloud
+func getAllOutboundContactlistfilterFn(ctx context.Context, p *outboundContactlistfilterProxy) (*[]platformclientv2.Contactlistfilter, error) {
+	var allContactlistfilters []platformclientv2.Contactlistfilter
+	const pageSize = 100
+
+	contactListFilters, _, err := p.outboundApi.GetOutboundContactlistfilters(pageSize, 1, true, "", "", "", "", "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get page of contact list filter: %v", err)
+	}
+
+	if contactListFilters.Entities == nil || len(*contactListFilters.Entities) == 0 {
+		return &allContactlistfilters, nil
+	}
+
+	for _, contactListFilter := range *contactListFilters.Entities {
+		allContactlistfilters = append(allContactlistfilters, contactListFilter)
+	}
+
+	for pageNum := 2; pageNum <= *contactListFilters.PageCount; pageNum++ {
+		contactListFilters, _, err := p.outboundApi.GetOutboundContactlistfilters(pageSize, pageNum, true, "", "", "", "", "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to get page of contact list filter: %v", err)
+		}
+
+		if contactListFilters.Entities == nil || len(*contactListFilters.Entities) == 0 {
+			break
+		}
+
+		for _, contactListFilter := range *contactListFilters.Entities {
+			allContactlistfilters = append(allContactlistfilters, contactListFilter)
+		}
+	}
+
+	return &allContactlistfilters, nil
+}
+
+// getOutboundContactlistfilterIdByNameFn is an implementation of the function to get a Genesys Cloud outbound contactlistfilter by name
+func getOutboundContactlistfilterIdByNameFn(ctx context.Context, p *outboundContactlistfilterProxy, name string) (id string, retryable bool, err error) {
+	contactListFilters, err := getAllOutboundContactlistfilterFn(ctx, p)
```

Comment:
It could be worth adding name as a parameter to this function and providing it here, the passing it to `GetOutboundContactlistfilters`. Filtering by name will save on API calls if there are loads of contact list filters in an org